codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/FluentClientDecorator.kt

```diff
@@ -128,6 +137,135 @@ data class ClientGenerics(
     }
 }
 
+class GenericFluentClient(codegenContext: CodegenContext) : FluentClientCustomization() {
+    val moduleUseName = codegenContext.moduleUseName()
+    override fun section(section: FluentClientSection): Writable {
+        return when (section) {
+            is FluentClientSection.FluentClientDocs -> writable {
+                val humanName = section.serviceShape.id.name
+                rust(
+                    """
+                    /// An ergonomic service client for `$humanName`.
+                    ///
+                    /// This client allows ergonomic access to a `$humanName`-shaped service.
+                    /// Each method corresponds to an endpoint defined in the service's Smithy model,
+                    /// and the request and response shapes are auto-generated from that same model.
+            ///"""
+                )
+                rust(
+                    """
+                    /// ## Constructing a Client
+                    ///
+                    /// To construct a client, you need a few different things:
+                    ///
+                    /// - A [`Config`](crate::Config) that specifies additional configuration
+                    ///   required by the service.
+                    /// - A connector (`C`) that specifies how HTTP requests are translated
+                    ///   into HTTP responses. This will typically be an HTTP client (like
+                    ///   `hyper`), though you can also substitute in your own, like a mock
+                    ///   mock connector for testing.
+                    /// - A "middleware" (`M`) that modifies requests prior to them being
+                    ///   sent to the request. Most commonly, middleware will decide what
```

Comment:
The wording of  `A "middleware" (`M`) that modifies requests prior to them being sent to the request.` sounds weird.